test/sync_tests/conflicting.test.js

```diff
@@ -0,0 +1,763 @@
+
+    const originalValue = note.title;
+    const dirtyValue = `${Math.random()}`;
+
+    await this.application.itemManager.changeNote(note.uuid, (mutator) => {
+      // modify this item locally to have differing contents from server
+      mutator.title = dirtyValue;
+      // Intentionally don't change updated_at. We want to simulate a chaotic case where
+      // for some reason we receive an item with different content but the same updated_at.
+      // note.updated_at = Factory.yesterday();
+    });
+
+    // Download all items from the server, which will include this note.
+    await this.application.syncService.clearSyncPositionTokens();
+    await this.application.syncService.sync({
+      ...syncOptions,
+      awaitAll: true,
+    });
+
+    // We expect this item to be duplicated
+    this.expectedItemCount++;
+    expect(this.application.itemManager.notes.length).to.equal(2);
+
+    const allItems = this.application.itemManager.items;
+    expect(allItems.length).to.equal(this.expectedItemCount);
+
+    const originalItem = this.application.itemManager.findItem(note.uuid);
+    const duplicateItem = allItems.find(
+      (i) => i.content.conflict_of === note.uuid
+    );
+
+    expect(originalItem.title).to.equal(dirtyValue);
+    expect(duplicateItem.title).to.equal(originalValue);
+    expect(originalItem.title).to.not.equal(duplicateItem.title);
+
+    const newRawPayloads = await this.application.storageService.getAllRawPayloads();
+    expect(newRawPayloads.length).to.equal(this.expectedItemCount);
+  });
+
+  it('should handle sync conflicts by duplicating differing data', async function () {
+    // create an item and sync it
+    const note = await Factory.createMappedNote(this.application);
+    await this.application.saveItem(note.uuid);
+    this.expectedItemCount++;
+
+    const rawPayloads = await this.application.storageService.getAllRawPayloads();
+    expect(rawPayloads.length).to.equal(this.expectedItemCount);
+    /** First modify the item without saving so that
+     * our local contents digress from the server's */
+    await this.application.changeItem(note.uuid, (mutator) => {
+      mutator.title = `${Math.random()}`;
+    });
+    await this.application.changeAndSaveItem(
+      note.uuid,
+      (mutator) => {
+        // modify this item to have stale values
+        mutator.title = `${Math.random()}`;
+        mutator.updated_at = Factory.yesterday();
+      },
+      undefined,
+      undefined,
+      syncOptions
+    );
+
+    // We expect this item to be duplicated
+    this.expectedItemCount++;
+    const allItems = this.application.itemManager.items;
+    expect(allItems.length).to.equal(this.expectedItemCount);
+
+    const note1 = this.application.itemManager.notes[0];
+    const note2 = this.application.itemManager.notes[1];
+    expect(note1.content.title).to.not.equal(note2.content.title);
+  });
+
+  it('basic conflict with clearing local state', async function () {
```

Comment:
What does this test?

Reply:
That a conflict is created. This is via `this.expectedItemCount++;`